Source/TestingServices/Runtime/TestingRuntime.cs

```diff
@@ -901,23 +904,33 @@ internal void Wait()
 
         #endregion
 
-        #region Timers
+        #region timers
 
         /// <summary>
-        /// Return the timer machine type
+        /// Creates a new timer that sends a <see cref="TimerElapsedEvent"/> to its owner machine after the specified
+        /// due time. The timer can be configured to periodically timeout, by specifying a non-negative period of time.
+        /// The timer accepts an optional payload to be used during timeout.
         /// </summary>
-        /// <returns></returns>
-        internal override Type GetTimerMachineType()
+        /// <param name="owner">The owner machine.</param>
+        /// <param name="dueTime">The amount of time to wait before sending the first timeout event.</param>
+        /// <param name="period">The time interval between timeout events. A negative value disables periodic timeout events.</param>
+        /// <param name="name">Optional unique name of the timer.</param>
+        /// <param name="payload">Optional payload of the timeout event.</param>
+        /// <returns>The machine timer.</returns>
+        internal override IMachineTimer CreateMachineTimer(Machine owner, TimeSpan dueTime, TimeSpan period, string name, object payload)
         {
-            var timerType = base.GetTimerMachineType();
-            if (timerType == null)
-            {
-                return typeof(Timers.ModelTimerMachine);
-            }
-
-            return timerType;
+            var mid = this.CreateMachineId(typeof(MockMachineTimer), name);
+            this.CreateMachine(mid, typeof(MockMachineTimer));
+            this.MachineMap.TryGetValue(mid, out Machine machine);
+            (machine as MockMachineTimer).Initialize(owner, dueTime, period, payload, this.Configuration.TimeoutDelay);
```

Comment:
Call a machine method directly seems wrong. Instead the `Initialize` should be called in the `OnEntry` method of the `start` state of `MockMachineTimer`.